tests/include/mir/test/doubles/mock_drm.h

```diff
@@ -66,20 +83,44 @@ class FakeDRMResources
 
     bool drm_setversion_called{false};
 private:
+    struct DRMObjectWithProperties
+    {
+        uint32_t object_type;
+        std::vector<uint32_t> ids;
+        std::vector<uint64_t> values;
+    };
+
+    struct DRMProperty
+    {
+        uint32_t id;
+        char name[DRM_PROP_NAME_LEN];
+    };
```

Comment:
Identical types already exist in `test_drm_mode_resources.cpp` (in a different scope) is there scope for consolidation?

Reply:
Yes, it was copied from there. I was going to do a follow up PR to update that test to the new API. I've done it now in this PR.